sklearn/tests/test_calibration.py

```diff
@@ -455,3 +456,89 @@ def test_calibration_attributes(clf, cv):
         classes = LabelBinarizer().fit(y).classes_
         assert_array_equal(calib_clf.classes_, classes)
         assert calib_clf.n_features_in_ == X.shape[1]
+
+
+@pytest.mark.parametrize('fit_params_as_list', [False, True])
+def test_calibration_with_fit_params(fit_params_as_list):
+    """Tests that fit_params are passed to the underlying base estimator
+
+    https://github.com/scikit-learn/scikit-learn/issues/12384
+
+    """
+    n_samples = 100
+    X, y = make_classification(n_samples=n_samples, n_features=6,
+                               random_state=42)
+
+    if fit_params_as_list:
```

Comment:
You can use `sklearn.utils._testing._convert_container` to avoid the if else:

```python
@pytest.mark.parametrize("fit_params_type", ["list", "array"])
...
...
fit_params = {
    "a": _convert_container(y), "y": _convert_container(y)
}
```

Reply:
:+1: